### DISCUSSION TITLE
How to set default string mutability to constant from python script?

### DISCUSSION THREAD

**Discussion by phix33** [NationalSecurityAgency/ghidra#5448] [Q&A] [answered]
I know that in the GUI I can right click on a string, select data, Default Settings and then change 'Mutability' from 'normal' to 'constant' but how would I do that in a python script?

While there's lots of great example scripts out there - for example, https://github.com/astrelsky/ghidra_scripts/blob/master/make_constants.py - I don't want to detract from that but 'wondered' if there was an easier way? Suspect my simple use case is probably different to the given example code (hence the query).

Thanks very much.

**Comment by astrelsky** [comment, marked as answer]
I _think_ there is a method in the `DataType` interface for getting the default settings. You could try getting the string data type instance for the current programs data type manager and set the setting as seen in the script.

**Reply by phix33** [reply]
Ahh that's the "trick" - thanks very much!

```
dtm = currentProgram.getDataTypeManager()
dtm.getDataType("/string").getDefaultSettings().setLong("mutability", 2)
```
